src/controllers/searchController.js

```diff
@@ -0,0 +1,132 @@
+import Models from '../db/models';
+import errorResponse from '../helpers/index';
+
+const { User, Article, Tag } = Models;
+
+/**
+ * @description class representing Article Controller
+ * @class SearchController
+ */
+class SearchController {
+  /**
+   * @description - This method is responsible for creating new articles
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   * @returns {object} - object representing response messages
+   * @memberof SearchController
+   */
+  static async searchByAuthor(request, response) {
+    const authorName = request.params.author;
+    try {
+      const author = await User.findOne({
+        where: {
+          username: authorName
+        },
+        include: [
+          {
+            model: Article,
+            as: 'articles',
+            attributes: ['slug', 'title', 'description', 'body', 'imgUrl']
+          }
+        ],
+        attributes: ['id', 'username', 'image', 'email', 'image']
+      });
+      if (author) {
+        if (author.dataValues.articles.length < 1) {
+          return response.status(404).json(errorResponse(['No article found']));
+        }
+        return response.status(200).json({
+          status: 'Success',
+          message: 'Articles found successfully',
+          author
+        });
+      }
+      return response.status(404).json(errorResponse(['Author not found']));
+    } catch (error) {
+      return response.status(500).json(errorResponse([error.message]));
+    }
+  }
+
+  /**
+   * @description - This method is responsible for creating new articles
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   * @returns {object} - object representing response message
+   * @memberof SearchController
+   */
+  static async searchByKeyword(request, response) {
+    const { keyword } = request.params;
+    try {
+      const articles = await Article.findAll({
+        where: {
+          $or: [
+            {
+              title: { $iLike: `%${keyword}%` }
+            },
+            {
+              description: { $iLike: `%${keyword}%` }
+            }
+          ]
+        },
+        include: [
+          {
+            model: User,
+            as: 'author',
+            attributes: ['id', 'username', 'email', 'image']
+          }
+        ],
+        attributes: ['slug', 'title', 'description', 'body', 'imgUrl', 'readtime']
+      });
+      if (articles.length < 1) {
```

Comment:
what happens when `articles.length === 0` versus `articles.length < 1`?

Reply:
same as above

Reply:
I think `articles.length < 1` is not necessary. Can the length here possibly be negative? Why not explicitly use `=== 0` or `!articles.length` to make it look elegant

Reply:
Hi Marcus, you're right. You can't have a negative length for the articles. But then `articles.length < 1` is less elaborate compared to `articles.length === 0`. I have refactored it though.